.github/actions/setup-builder/action.yaml

```diff
@@ -42,6 +42,8 @@ runs:
         "${RETRY[@]}" rustup component add rustfmt
     - name: Configure rust runtime env
       uses: ./.github/actions/setup-rust-runtime
+    - name: Setup Rust cache
+      uses: Swatinem/rust-cache@v2
```

Comment:
This is basically my only concern with the PR (using a third-party github action)

I took a brief look through https://github.com/Swatinem/rust-cache and it seems like it does non trivial work (there is a lot of code) and it says:

> It is also most effective for repositories with a Cargo.lock file. Library repositories with only a Cargo.toml file have limited benefits, as cargo will always use the most up-to-date dependency versions, which may not be cached.

I made an experimental PR to see how much this particular action saves in terms of time:
- https://github.com/apache/datafusion/pull/13889

If it doesn't save a huge amount of time, I think we should consider not using it

Reply:
I think the main benefit of it vs the normal GH cache is that it only caches dependencies, not the DF crates. That means the caches are smaller and are more likely to not churn the 10GB limit for the GH cache. 

From my tests the caches are effective but yes, it may churn a bit as dependencies are updated.

Reply:
![image](https://github.com/user-attachments/assets/b0a716a2-6988-4784-8a2d-4ae3a6e0d10c)

Essentially the same as the 'cold' test I did (cache unpopulated). Up to you, I have no objections either way.

Reply:
Let's keep things simple and remove the attempted caching

I think I may also be somewhat biased as really earler versions of the arrow and datafusion CI jobs tried to use the caching (not this github action, but some other one) and we eventually found that due to how picky rust was, it was almost impossible to get the compiler to re-use the intermediate stuff in `target` so all the caching was simply wasted.

I also found the same thing in our .circleci jobs in Influx (aka that caching the target directly was never effetive)